components/p3a/star_randomness_points.h

```diff
@@ -30,48 +30,34 @@ struct P3AConfig;
 // server in order to receive randomness point data for STAR measurements.
 class StarRandomnessPoints {
  public:
-  using RandomnessDataCallback = base::RepeatingCallback<void(
-      std::string metric_name,
-      MetricLogType log_type,
-      uint8_t epoch,
-      ::rust::Box<constellation::RandomnessRequestStateWrapper>
-          randomness_request_state,
+  using RandomnessDataCallback = base::OnceCallback<void(
       std::unique_ptr<rust::Vec<constellation::VecU8>> resp_points,
       std::unique_ptr<rust::Vec<constellation::VecU8>> resp_proofs)>;
 
   StarRandomnessPoints(
       scoped_refptr<network::SharedURLLoaderFactory> url_loader_factory,
-      RandomnessDataCallback data_callback,
       const P3AConfig* config);
   ~StarRandomnessPoints();
   StarRandomnessPoints(const StarRandomnessPoints&) = delete;
   StarRandomnessPoints& operator=(const StarRandomnessPoints&) = delete;
 
   void SendRandomnessRequest(
-      std::string metric_name,
       MetricLogType log_type,
       uint8_t epoch,
       StarRandomnessMeta* randomness_meta,
-      ::rust::Box<constellation::RandomnessRequestStateWrapper>
-          randomness_request_state,
-      const rust::Vec<constellation::VecU8>& rand_req_points);
+      const rust::Vec<constellation::VecU8>& rand_req_points,
+      RandomnessDataCallback callback);
 
  private:
-  void HandleRandomnessResponse(
-      std::string metric_name,
-      MetricLogType log_type,
-      uint8_t epoch,
-      StarRandomnessMeta* randomness_meta,
-      ::rust::Box<constellation::RandomnessRequestStateWrapper>
```

Comment:
perhaps we can move lib.rs.h include to .cc now

Reply:
Unfortunately not yet. This file still uses types from rust cxx reflection in the public definition of `StarRandomnessPoints`.